src/abstract_gp/finite_gp.jl

```diff
@@ -14,16 +14,24 @@ function FiniteGP(f::AbstractGP, x::AbstractVector, σ²::AbstractVector{<:Real}
     return FiniteGP(f, x, Diagonal(σ²))
 end
 
+const default_σ² = 1e-18
+
 FiniteGP(f::AbstractGP, x::AbstractVector, σ²::Real) = FiniteGP(f, x, Fill(σ², length(x)))
 
-FiniteGP(f::AbstractGP, x::AbstractVector) = FiniteGP(f, x, 1e-18)
+FiniteGP(f::AbstractGP, x::AbstractVector) = FiniteGP(f, x, default_σ²)
+
+function FiniteGP(
+    f::AbstractGP, X::AbstractMatrix;
+    obsdim::Int = KernelFunctions.defaultobs
+    )
+    return FiniteGP(f, X, default_σ²; obsdim=obsdim)
+end
 
 function FiniteGP(
-    f::AbstractGP,
-    X::AbstractMatrix;
+    f::AbstractGP, X::AbstractMatrix, σ²::Real;
```

Comment:
Maybe also just use
```suggestion
    f::AbstractGP, X::AbstractMatrix, σ²::Real = default_σ²;
```
here and remove the block above?

Reply:
I wanted to keep the style from before. @willtebbutt was there any reason to have it this way?

Reply:
Yeah, I would just use a default argument and retain a single method involving a matrix, rather than having two.

It would also make sense to drop the type constraint here, since the noise variance doesn't have to be `Real`.